rust/main/agents/relayer/src/msg/metadata/base_builder/validator_announced_storages/tests.rs

```diff
@@ -0,0 +1,147 @@
+use async_trait::async_trait;
+use mockall::mock;
+
+use hyperlane_base::cache::LocalCache;
+use hyperlane_core::{
+    Announcement, ChainCommunicationError, ChainResult, HyperlaneChain, HyperlaneContract,
+    HyperlaneDomain, HyperlaneProvider, SignedType, TxOutcome, ValidatorAnnounce, H256, U256,
+};
+
+use super::*;
+
+// Mock implementation for ValidatorAnnounce
+mock! {
+    #[derive(Debug)]
+    pub ValidatorAnnounceMock {}
+
+    #[async_trait]
+    impl ValidatorAnnounce for ValidatorAnnounceMock {
+        async fn get_announced_storage_locations(
+            &self,
+            validators: &[H256],
+        ) -> ChainResult<Vec<Vec<String>>>;
+
+        async fn announce(&self, announcement: SignedType<Announcement>) -> ChainResult<TxOutcome>;
+
+        async fn announce_tokens_needed(&self, announcement: SignedType<Announcement>) -> Option<U256>;
+    }
+
+    impl HyperlaneContract for ValidatorAnnounceMock {
+        fn address(&self) -> H256;
+    }
+
+    impl HyperlaneChain for ValidatorAnnounceMock {
+        fn domain(&self) -> &HyperlaneDomain;
+        fn provider(&self) -> Box<dyn HyperlaneProvider>;
+    }
+}
+
+#[tokio::test]
```

Comment:
would probably be good to test that different domains don't overlap in the cache

Reply:
Added